### ISSUE TITLE
pycanberra ignores system configured sound theme

### ISSUE THREAD

**Issue by elad661** [totdb/pycanberra#1]
From IRC:

> <elad> hey, around? I noticed that changing sound theme setting in GSettings affects canberra-gtk-play as expected, but doesn't affect things that use libcanberra directly, eg. pycanberra: It's still looking for the freedesktop theme according to strace. any  idea why?
> <poettering> well, only canberra-gtk binds to gsettings
> <elad> ah, okay, so I need to manually specify the theme in pycanberra apps?
> <poettering> yeah, i fear
> <elad> okay, thanks
> <poettering> the idea is basically that for canberra-gtk they data comes from gsettings, but for a (theoretic) canberra-kde it comes from some kde specific source

I suggest that pycanberra will add the code to read the configured sound theme from gsettings and use it. I don't know much about ctypes or how to fix it myself, but here's some of the work:

``` python
#put that in your imports area at the top
try:
    from gi.repository import Gio
    has_gio = True
except ImportError:
    has_gio = False

#And this when you actually play the sounds or create the Canberra object (your choice)
if has_gio:
    settings = Gio.Settings.new('org.gnome.desktop.sound')
    theme = settings.get_string('theme-name')

#Insert magic to actually tell libcanberra to use the theme you found here
```

**Comment by elad661** [comment]
Researching the docs, seems that you need to use [`CA-PROP-CANBERRA-XDG-THEME-NAME`](http://0pointer.de/lennart/projects/libcanberra/gtkdoc/libcanberra-canberra.html#CA-PROP-CANBERRA-XDG-THEME-NAME:CAPS). I'm not familiar enough with ctypes to write my own patch yet, tho.

**Comment by bochecha** [comment]
Elad, can you try the following code:

```
>>> import pycanberra
>>> canberra = pycanberra.Canberra()
>>> canberra.play(1, pycanberra.CA_PROP_EVENT_ID, "dialog-error", pycanberra.CA_PROP_MEDIA_ROLE, "error", pycanberra.CA_PROP_CANBERRA_XDG_THEME_NAME, "mytheme", None)
```

Of course, replace `"mytheme"` by the actual nameof your theme.

Does that work?

**Comment by elad661** [comment]
I can confirm this works, yes.

**Comment by pbor** [comment]
I do not think this should go in pycanberra: pycanberra is a thin wrapper that should expose the same features of libcanberra without adding logic on its own. This logic belongs to the app (it's just a couple of lines) or to an higher level lib
